system/jlib/jmetrics.cpp

```diff
@@ -390,3 +401,20 @@ std::shared_ptr<GaugeMetricFromCounters> hpccMetrics::registerGaugeFromCountersM
     queryMetricsManager().addMetric(pMetric);
     return pMetric;
 }
+
+
+std::shared_ptr<HistogramMetric> registerHistogramMetric(const char *name, const char* desc, StatisticMeasure units, const std::vector<BucketDef> &bucketDefs, const MetricMetaData &metaData)
+{
+    std::shared_ptr<HistogramMetric> pMetric = std::shared_ptr<HistogramMetric>(new HistogramMetric(name, desc, units, bucketDefs, metaData));
+    queryMetricsManager().addMetric(pMetric);
+    return pMetric;
+}
+
+
+std::shared_ptr<ScaledHistogramMetric> registerScaledHistogramMetric(const char *name, const char* desc, StatisticMeasure units, const std::vector<BucketDef> &bucketDefs,
+                                                                               double scaleFactorToReport, const MetricMetaData &metaData)
```

Comment:
The name of the parameter scaleFactorToReport should be made consistent with limitsToMeasurementUnitsScaleFactor, especially because they have the opposite meaning.